File: jas_ocaml/lib/tools/interior_selection_tool.ml
(** Interior Selection tool: marquee select that picks groups as units. *)

class interior_selection_tool = object
  inherit Selection_tool.selection_tool_base

  method private check_handle_hit (_ctx : Canvas_tool.tool_context) (_x : float) (_y : float) = false
  method private select_rect (ctx : Canvas_tool.tool_context) x y w h ~extend =
    ctx.controller#group_select_rect ~extend x y w h
end

File: JasSwift/Sources/Tools/InteriorSelectionTool.swift
import AppKit
import Foundation

// Interior Selection tool — marquee select that picks groups as units.

class InteriorSelectionTool: SelectionToolBase {
    override func selectRect(_ ctx: ToolContext, x: Double, y: Double, w: Double, h: Double, extend: Bool) {
        ctx.controller.groupSelectRect(x: x, y: y, width: w, height: h, extend: extend)
    }
}
